File: app/src/main/java/com/medziku/plantwatering/data/parts/AmperageSensorImpl.java
package com.medziku.plantwatering.data.parts;

import android.os.Parcel;
import android.os.Parcelable;

import com.medziku.plantwatering.station.parts.AmperageSensor;


public class AmperageSensorImpl extends AmperageSensor implements Parcelable {

    public AmperageSensorImpl(String name, float amperage) {
        super(name, amperage);
    }

    private AmperageSensorImpl(Parcel in) {
        super(in.readString(), in.readFloat());
    }

    public static final Creator<AmperageSensorImpl> CREATOR = new Creator<AmperageSensorImpl>() {
        @Override
        public AmperageSensorImpl createFromParcel(Parcel in) {
            return new AmperageSensorImpl(in);
        }

        @Override
        public AmperageSensorImpl[] newArray(int size) {
            return new AmperageSensorImpl[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(getName());
        dest.writeFloat(getAmperage());
    }
}

File: android-app/src/main/java/com/medziku/plantwatering/data/parts/AmperageSensorImpl.kt
package com.medziku.plantwatering.data.parts

import android.os.Parcel
import android.os.Parcelable

import com.medziku.plantwatering.station.parts.AmperageSensor

class AmperageSensorImpl : AmperageSensor, Parcelable {

    constructor(name: String, amperage: Float) : super(name, amperage)

    private constructor(`in`: Parcel) : super(`in`.readString(), `in`.readFloat())

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(name)
        dest.writeFloat(amperage)
    }

    companion object {

        @JvmField
        val CREATOR: Parcelable.Creator<AmperageSensorImpl> = object : Parcelable.Creator<AmperageSensorImpl> {
            override fun createFromParcel(`in`: Parcel): AmperageSensorImpl {
                return AmperageSensorImpl(`in`)
            }

            override fun newArray(size: Int): Array<AmperageSensorImpl?> {
                return arrayOfNulls(size)
            }
        }
    }
}
